src/cpp/session/modules/SessionConsole.R

```diff
@@ -14,3 +14,71 @@
 #
 #
 
+.rs.addJsonRpcHandler("console_follow_hyperlink", function(url, text, params)
+{
+  if (!grepl("://", url)) {
+    stop("missing scheme")
+  }
+
+  if (identical(params, "")) {
+    params <- list()
+  } else {
+    params <- strsplit(params, ":")
+
+    names  <- sapply(params, function(p) sub("=.+$", "", p))
+    params <- lapply(params, function(p) sub("^[^=]+=", "", p))
+    names(params) <- names
+  }
+  
+  scheme <- sub("://.*$", "", url)
+  handler <- switch(scheme, 
+    # rstatshelp:://stats/rnorm
+    # rstatshelp:://stats::rnorm
+    rstatshelp = function(url) {
+      url <- sub("rstatshelp://", "", url)
+      parts <- strsplit(url, "/")[[1L]]
+
+      if (length(parts) == 2L) {
```

Comment:
What if you want to link to an arbitrary help document; e.g. a vignette?

Reply:
Maybe `rstudio:help` and `rstudio:vignette` would be better scheme names? (inspiration from https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app)

Reply:
Maybe the `http://localhost:8787` prefix also deals with this.

Reply:
@hadley You mean that `rstudio` would be the scheme name? And then have something like
```
rstudio:help?topic=dplyr::filter
rstudio:vignette?name=vctrs::type-size
```
or
```
rstudio:help/dplyr/filter
rstudio:vignette/vctrs/type-size
```
?

@romainfrancois I am probably missing something, but is the 8787 port number fixed? Will `localhost:8787` issues with personal firewalls?

Reply:
`tools::startDynamicHelp(NA) ` starts the HTTP help server, which seems overkill for just some CLI output.

Reply:
If within rstudio, we can use `.rs.httpdPort()` and chances are the server is already running, `cli` could do something else otherwise. Can also make the case that the link is likely to be clicked and that would need the server to be ready then ...

Reply:
True, but most of the time the link will not be clicked, so it seems better to start the server when it is clicked.